.travis.yml

```diff
@@ -0,0 +1,66 @@
+    - secure: "gOWWEMWMUnNWJmwH25wEzjcbOMhA6E/WHyNoI1QclNZFrHtUhhZXYO53lTWtAgvRZf9D0ouPmwAcE7Hgbjh/ql3ERWpsU8W0XiolCopd5zMfYp4gpxiQVdiYTyEW8WYv8mrtwQ/lHzWSij58MAsgBDML/IYdgHHvl7Ty8b2tGE3dUd+wrLsE8nZ3XohyUELnqfNAC5+qPOv3vWSL3cAVOSSAkWsrsowbpnbprpNmV18HO0dZZno3cGj3Sj9KOUTE6ceS1TSyz9oyzyZ3zp63IUkISGS6hOcMmQEU1toixO80tQG447mkRYAZlsXo6z3rjYNNvxVZsmKFxreYeBZGbOJeiW7fI8ZBin2fhuRoERYP84psB1IFEvOuBGyrHZiScZgmFPF0r75pp8we7w/3fBQq35luCnL5hJQ13V28JFDyHYUPAm5+alzUIA2hOBjYBtGVhSSevLlpQrLUdGmxqZFWumo2qXXWxsabsbwWG4oK3ixCJP1ET0rkls5BmLr1MjWLJN30miekP54O/gk23yWH0XviAbs8MOUY+Mel6TiwEF2N1X7UdjfMSrcnKknv78dfvoAFByqlT54oiZv6yzpb7cyFuQfqRWBiMDKulu6u819F5UQzE7Z+rVOQnXglPkrwuJu8ky8Lrsdd2Lfrbc6Zaz3w5+Tcixux1jbVvhQ="
+
+services:
+  - docker
+
+before_script:
+  - curl https://cli-assets.heroku.com/install.sh | sh
+  - docker login --username $docker_login --password $docker_password
+
+script:
+
+  # Docker Build the Image
+  - docker build --target test --tag todoapp:test .
+
+  # Docker Run Unit Test
+  - echo "Running Unit Tests"
+  - docker run todoapp:test test_view_model.py
+
+  # Docker Run Integration tests 
+  - echo "Running Integration Tests"
+  #- docker run todoapp:test test_todoapp.py
+
+  # Docker Run E2E 
+  - echo "Running E2E Tests"
+  - docker run -e trello_key=$trello_key -e trello_token=$trello_token -e trello_boardid=$trello_boardid -e todo_list_id=$todo_list_id -e doing_list_id=$doing_list_id -e done_list_id=$done_list_id todoapp:test test_trello.py
+
+  # Docker Build and Push 
+  - docker build --target production --tag $docker_login/todoapp:latest .
+  - docker push $docker_login/todoapp:latest
+
+  # Docker Pull latest image & Tag
+  - echo "Pull image from Dockerhub and Tag"
+  - docker pull $docker_login/todoapp:latest     
+  - docker tag $docker_login/todoapp:latest registry.heroku.com/rajrahmantodoapp/web
+
+  # Login to Heroku
+  - echo "Login to Heroku"
+  - echo $HEROKU_API_KEY | docker login --username=$heroku_login --password-stdin registry.heroku.com
+
+  # Push image to Heroku
+  - echo "Push image to Heroku"
+  - docker push registry.heroku.com/rajrahmantodoapp/web
+
+# Release image in Heroku
+deploy:
+  provider: script
+  script: 
+      - echo "Heroku Release to web"
+      - heroku container:release web --app rajrahmantodoapp
+  on:
+      all_branches: true
```

Comment:
We should remember to update this when you're ready to merge so this only runs on master